docs/guides/auth.md

```diff
@@ -180,7 +180,7 @@ var client = new Greeter.GreeterClient(channel);
 ####Base case - No encryption/authentication
```

Comment:
@LisaFC Yes, my plan was to do pretty much what the your last paragraph said and re-organize sections by language.
The subsections should be:
- no authentication cleartext
- server-only authentication with SSL/TLS
- client authentication with Google over server-authenticated SSL/TLS

Sounds good? We could add a section later for mutual authentication with SSL/TLS with SSL client certs but the setup is slightly more complicated and that would just add noise at this point.